sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1730,6 +1730,14 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     def assertExecutionId: UserDefinedFunction = udf(AssertExecutionId.apply _)
     spark.range(10).select(assertExecutionId($"id")).localCheckpoint(true)
   }
+
+  test("implicit encoder for LocalDate and Instant") {
+    val localDate = java.time.LocalDate.of(2019, 3, 30)
+    assert(spark.range(1).map { _ => localDate }.head == localDate)
```

Comment:
Nit: use `===`

Reply:
Why is it better than `==` in this case? 
1. With `==`:
```scala
val localDate = java.time.LocalDate.of(2019, 3, 30)
assert(spark.range(1).map { _ => 1 }.head == localDate)
```
```
1 did not equal 2019-03-30
ScalaTestFailureLocation: org.apache.spark.sql.DatasetSuite at (DatasetSuite.scala:1736)
Expected :2019-03-30
Actual   :1
<Click to see difference>

org.scalatest.exceptions.TestFailedException: 1 did not equal 2019-03-30
```
2. With `===`:
```scala
assert(spark.range(1).map { _ => 1 }.head === localDate)
```
```
1 did not equal 2019-03-30
ScalaTestFailureLocation: org.apache.spark.sql.DatasetSuite at (DatasetSuite.scala:1736)
Expected :2019-03-30
Actual   :1
<Click to see difference>

org.scalatest.exceptions.TestFailedException: 1 did not equal 2019-03-30
```
@srowen Is it worth to rebuild and wait for another 4 hours?

Reply:
Hm, I thought `==` would just trigger `.equals()` and so the scalatest macro could only report "assertion failed". I guess it's smarter than I thought as it can look at the content of the assertion and be smart. OK yeah leave that.